### PR TITLE
Update tests for new EIP2333 (bls v2 compliant HKDF_mod_r)

### PR DESCRIPTION
**Pull request by CarlBeek**
EIP 2333 [updated](https://github.com/ethereum/EIPs/pull/2744/files) to support BLSv2's KeyGen style HKDF_mod_r. This PR makes this tool compliant.

### CODE REVIEW

File: eth2deposit/key_handling/key_derivation/tree.py [hunk 1]
```diff
@@ -37,8 +37,14 @@ def _parent_SK_to_lamport_PK(*, parent_SK: int, index: int) -> bytes:
     return compressed_PK
 
 
-def _HKDF_mod_r(*, IKM: bytes) -> int:
-    okm = HKDF(salt=b'BLS-SIG-KEYGEN-SALT-', IKM=IKM, L=48)
+def _HKDF_mod_r(*, IKM: bytes, key_info: bytes=b'') -> int:
```
Comment (hwwhww) [inline eth2deposit/key_handling/key_derivation/tree.py:40]: What do you think about adding a test vector of that `key_info != b''`?

File: eth2deposit/key_handling/key_derivation/tree.py [hunk 2]
```diff
@@ -37,8 +37,14 @@ def _parent_SK_to_lamport_PK(*, parent_SK: int, index: int) -> bytes:
...
+    L = 48
```
Comment (hwwhww) [inline eth2deposit/key_handling/key_derivation/tree.py:41]: Maybe add a note:
```suggestion
    L = 48  # `ceil((3 * ceil(log2(r))) / 16)`, where `r` is the order of the BLS 12-381 curve
```

File: eth2deposit/key_handling/key_derivation/tree.py [hunk 3]
```diff
@@ -37,8 +37,14 @@ def _parent_SK_to_lamport_PK(*, parent_SK: int, index: int) -> bytes:
...
+    okm = HKDF(
+        salt=b'BLS-SIG-KEYGEN-SALT-',
+        IKM=IKM + b'\x00',
```
Comment (hwwhww) [inline eth2deposit/key_handling/key_derivation/tree.py:44]: By the way, do you know what's the original rationale of adding `I2OSP(0, 1)`?

IMHO it's a bit confusing of what variable `IKM` should be, although I understand `IKM` in different in BLS v2 spec and RFC5869. Maybe add some comments for it,
```suggestion
        IKM=IKM + b'\x00',  # add postfix `I2OSP(0, 1)`
```

File: eth2deposit/key_handling/key_derivation/tree.py (hunk 1 above)
Comment (CarlBeek) [reply eth2deposit/key_handling/key_derivation/tree.py:40]: There currently aren't any tests for this so I tested against py_ecc.

**Pull request merged**